controllers/image_generator.go

```diff
@@ -59,6 +59,8 @@ const (
 	LibvirtProvider               = "libvirt"
 	peerpodsImageJobsPathLocation = "/config/peerpods/podvm"
 	azureImageGalleryPrefix       = "PodVMGallery"
+	preConfigJobName              = "osc-podvm-pre-configuration"
```

Comment:
On a generic note, I'm really not in favour of adding more jobs to the already complicated image generation code. 
Can't the jobs be run in a standalone fashion  (admin runs oc apply -f <job.yaml> or a script) to start with without code integration ?

Reply:
@bpradipt  If we provide builder image with these scripts, we can run this job by admin.
We thought creation/deletion of configMaps, Secrets and Libvirt Pool/volume can be automated with OSC. We tried to synchronized with Image Job and code.
We enhanced this to be scalable  for other providers also

Reply:
Can't the configmap and secret updated be de-coupled from image generation ?

Reply:
@bpradipt  Certainly possible to de-couple configmap and secret from image generation
This can be implemented as below:

File config/peerpods/podvm/libvirt-podvm-secret.yaml contains all user Input.
Add new Key/Flag in the config/peerpods/podvm/libvirt-podvm-secret.yaml file as below:
`IMAGE_GEN ="YES"`
This is an option parameter, by default it can be considered as `Yes` . If Admin user doesn't wish to generate image, then it can be set to "NO". Then it will not create image config map and pool, vol.
By default, it can create peer-pods-cm configmap and secrets.

If this option is fine, we can implement and drop that code changes

Reply:
Let's hear what other maintainers have to say before you make any changes. 

The related issue is quite old and lots have changed since then. So I think some discussion is needed on the linked issue - https://github.com/openshift/sandboxed-containers-operator/issues/435

Reply:
Thanks @bpradipt  

Please let me know where I need to initiate this discussion. I will initiate the discussion